api-service/crowdsource-webapp/src/app/relative-toxicity-answers-page/relative-toxicity-answers-page.component.ts

```diff
@@ -0,0 +1,69 @@
+import { Component, OnInit, Input, ViewChild } from '@angular/core';
+import { ActivatedRoute, Params, ParamMap, Router } from '@angular/router';
+import { AnswerSummary, CrowdsourceApiService } from '../crowdsource-api.service';
+import { RelativeToxicityAnswer } from '../relative-toxicity-job/relative-toxicity-job.component';
+
+interface QuestionsToAnswersMap {
+  [questionId: string]: AnswerSummary<RelativeToxicityAnswer>[];
+}
+
+/**
+ * Component that displays tea answers for examples in a relative rating task.
+ */
+@Component({
+  selector: 'app-relative-toxicity-answers-page',
+  templateUrl: './relative-toxicity-answers-page.component.html',
+  styleUrls: ['./relative-toxicity-answers-page.component.css'],
+})
+export class RelativeToxicityAnswersPageComponent implements OnInit {
+  clientJobKey: string;
+  answers: AnswerSummary<RelativeToxicityAnswer>[] = [];
+  questionsToAnswersMap: QuestionsToAnswersMap = {};
+  questionIds: string[] = [];
+  debugMode = false;
+
+  constructor(
+    private route: ActivatedRoute,
+    private crowdSourceApiService: CrowdsourceApiService) { }
+
+  ngOnInit(): void {
+    this.route.paramMap.subscribe((params: ParamMap) => {
+      this.updateIds(params);
+      this.getAnswers();
+    });
+  }
+
+  updateIds(params: ParamMap): void {
+    const clientJobKey = params.get('clientJobKey');
+    if (clientJobKey === null) {
+      throw new Error('clientJobKey must be specified.');
+    }
+    this.clientJobKey = clientJobKey;
+  }
+
+  getAnswers() {
+    if (!this.clientJobKey) {
+      throw new Error('Can\'t fetch answers without a client job key');
+    }
+    this.crowdSourceApiService.getAnswers(this.clientJobKey)
+        .subscribe(
+          (data: AnswerSummary<RelativeToxicityAnswer>[]) => {
+            console.log(data);
+            this.answers = data;
+            this.questionsToAnswersMap = {};
+            this.questionIds = [];
+            this.answers.forEach((answer, index) => {
+              if (!this.questionsToAnswersMap[answer.question_id]) {
+                this.questionsToAnswersMap[answer.question_id] = [];
+                this.questionIds.push(answer.question_id);
```

Comment:
I'm a bit worried about pushing all `question_id`s, wont this result in duplicates, and then in the display we'll show the same graph many times... ? Maybe add this to a set, and then turn the set to an array after to avoid dups?

Reply:
It shouldn't, I don't think...I basically have it set up so that the ids list is a set of the keys of the map (the if condition ensures they only get added to the list if they're not yet in the map, so each id should only get added once). I did that instead of Object.keys() to simplify data binding, but if it's confusing I can change it.

Reply:
Just add a comment here to explain should be fine. Presumably it's assuming something coming back from the API is a set then? A comment explaining here (and/or where variable is defined should be enough) will help.

Reply:
Done.